### PR TITLE
feat: Socket.IO connects with the jwt token

### LINKED ISSUE #318: feat: SocketIO 連線時帶 token 提供驗證
image

### PR DESCRIPTION
**Pull request by allyusd**
## Why need this change? / Root cause:

- Socket.IO connects with the jwt token

## Changes made:

- When the user's token changes, Socket.IO will automatically reconnect using the new token.
- Socker.IO dev server can get token

## Test Scope / Change impact:

-

## Issue

- Close #318

## Note

原本的 Socket.IO 是固定參數，這次需求改為動態帶 auth token

### Reference

https://socket.io/docs/v4/middlewares/#sending-credentials

### CODE REVIEW

**Review by Yuwen-ctw — APPROVED** [review]
LGTM!

File: pages/api/mock/socketio.ts [hunk 1]
```diff
@@ -32,6 +32,11 @@ const socketio = async (req: NextApiRequest, res: NextApiResponseServerIO) => {
       addTrailingSlash: false,
     });
 
+    io.use((socket, next) => {
+      const token = socket.handshake.auth.token;
```
Comment (JohnsonMao) [inline pages/api/mock/socketio.ts:36]: 這邊是要測試當沒有 token 時，進行拋錯嗎

**Review by JohnsonMao — APPROVED** [review]
LGTM!

File: pages/api/mock/socketio.ts (hunk 1 above)
Comment (allyusd) [reply pages/api/mock/socketio.ts:36]: 這邊一方面是確認範例方法有效，Server 端可以透過這段程式拿到 token。
另一方面是保留給 mock 後面實作，因為還沒有規劃 token 後續要做什麼，
所以只有該行程式碼。

有發現觸發  Code Smell,  useless variable, 但是不確定怎麼處理比較好？
註解掉該行？下 eslint disable next line 不知道對 Code Smell 有沒有效，
或者其它的用途？丟到 unit test 使用？或者像你說的，直接寫判斷拋錯？

Comment (JohnsonMao) [reply pages/api/mock/socketio.ts:36]: 通常都直接寫判斷拋錯，或是註解提醒以後可以實作這裡的 edge case 來更完善程式碼，目前這是在 mock api，所以目前這樣也無傷大雅

Comment (allyusd) [reply pages/api/mock/socketio.ts]: ok, 那我實作 error handler
